### PR TITLE
Added preview for attachments

### LINKED ISSUE #740: Build preview for Attachments
Attachment preview feature requirements:
1. Show previews for image & video attachments in a modal
2. The user should be able to navigate through previews
3. Show prev/next navigation icons on the preview attachments modal
4. Add keyboard shortcuts for:
   a) `Escape` key to close the modal
   b) `Left` & `Right` arrow keys for navigating through previews

_Originally posted by @abilash-sajeev in https://github.com/bigbinary/neeto-desk-web/issues/15654#issuecomment-1505102448_

---

@gaagul _a let's take this up. This is the full context: https://vimeo.com/816885756/a14b4f2047

Reference to the previous implementation:
image

### PR DESCRIPTION
**Pull request by gaagul**
- Fixes #740 

**Description**

- Added: preview for attachments

**Checklist**

~~- [ ] I have made corresponding changes to the documentation.~~
~~- [ ] I have updated the types definition of modified exports.~~
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

**Reviewers**

@AbhayVAshokan _a

### CODE REVIEW

**Comment by gaagul** [comment]
@AbhayVAshokan _a Please review.

File: src/components/Attachments/PreviewModal.jsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import React, { useRef } from "react";
+
+import saveAs from "file-saver";
+import { findIndexBy } from "neetocommons/pure";
+import { Download, LeftArrow, RightArrow } from "neetoicons";
```
Comment (AbhayVAshokan) [inline src/components/Attachments/PreviewModal.jsx:5]:
```suggestion
import { Download, Left, Right } from "neetoicons";
```

`Left` and `Right` icons have better aesthetics.

**Review by AbhayVAshokan — CHANGES_REQUESTED** [review]
@gaagul_a 
Also check: https://vimeo.com/819069679/69662f01fd?share=copy

File: src/components/Attachments/Attachment.jsx [hunk 2]
```diff
@@ -140,35 +141,43 @@ const Attachment = ({
           </>
         ) : (
           <>
-            <FileIcon fileName={attachment.filename} />
-            <Tooltip content={attachment.filename} position="top">
-              <Typography style="body2">{attachment.filename}</Typography>
-            </Tooltip>
+            <div
+              className="ne-attachments__attachment__preview-wrapper"
```
Comment (AbhayVAshokan) [inline src/components/Attachments/Attachment.jsx:145]:
```suggestion
              className="ne-attachments__preview-wrapper"
```

This class name should be good enough. Make similar changes to the sub classes.

**Review by AbhayVAshokan — APPROVED** [review]
Nice work 👌. Please merge after making this change.

**Pull request merged**